### ISSUE TITLE
Add startup script support for Google Cloud Engine

### ISSUE THREAD

**Issue by tantrev** [gc3-uzh-ch/elasticluster#237]
I was just looking at the ["Source code for elasticluster.providers.gce"](http://elasticluster.readthedocs.org/en/latest/_modules/elasticluster/providers/gce.html) page and it appears that there isn't any native support for [startup scripts](https://cloud.google.com/compute/docs/startupscript).  Such functionality would, of course, be very much appreciated. :)

**Comment by tantrev** [comment]
Looks like there's a [duplicate issue ](https://github.com/googlegenomics/elasticluster/issues/5)over in the GCE fork of Elasticluster.

**Comment by ndmckinley** [comment]
FYI: this does seem to be supported, under the name `image_userdata` (which is an old name for a startup script).

**Comment by riccardomurri** [comment]
It's definitely intended to be supported -- but I think somebody complained on the mailing-list that the current implementation doesn't really work?

**Comment by riccardomurri** [comment]
Indeed, if I use this cluster template, no file `hello.txt` gets created (checked `/`,  `/root`, and `/home/ubuntu`):

    [cluster/u16g]
    cloud=google
    setup=test

    login=google
    image_id=ubuntu-1604-xenial-v20190212
    flavor=g1-small
    security_group=default

    master_nodes=1
    ssh_to=master

    image_userdata=#!/bin/sh
            echo hello > "$HOME/hello.txt"

**Comment by ndmckinley** [comment]
Interesting...  Let me look into that.  It looks correctly implemented at first blush, but I'll look a bit deeper.

**Comment by ndmckinley** [comment]
![image](https://user-images.githubusercontent.com/605324/53829776-c0aeab80-3f35-11e9-90e7-84337bb39363.png)

generates

![image](https://user-images.githubusercontent.com/605324/53829807-d58b3f00-3f35-11e9-9019-f05b7d1fffaa.png)

which includes

![image](https://user-images.githubusercontent.com/605324/53829824-e0de6a80-3f35-11e9-9e62-4bae51cb2106.png)

which executes and functions correctly:

![image](https://user-images.githubusercontent.com/605324/53829857-f5226780-3f35-11e9-81d3-9bf82dc927ce.png)

This makes me wonder what issues folks are having - I'd be happy to help them debug!  One notable difference between our configs is that yours has a multiline string - I know nothing about ini multiline string parsing.  The feature on the GCE side does seem to be correctly implemented, though!

**Comment by ndmckinley** [comment]
Oh, also the `$HOME` environment variable - but that should certainly be set to _something_.  I wonder what user the script runs as!

**Comment by ndmckinley** [comment]
Ah, just experimented - `$HOME` is unset, and `whoami` is `root`.  If `/hello.txt` didn't get run when you did that, I suspect it's because of multiline string parsing issues - if you check the cloud console page's "custom metadata" section we could find out for sure.
